### PR TITLE
Fix for - Running func new in an empty directory creates incorrect fi…

### LINKED ISSUE #4581: Running `func new` in an empty directory creates incorrect files for dotnet
### Version

4.1.0

### Description

If you run func new directly in an empty directory, a host.json file with bundle configuration is created even before you select the language. In that case if you select dotnet inproc or dotnet isolated you end up getting the below error.

### Steps to reproduce

1. Run `func new` in an empty directory
2. Select dotnet isolated

```
Use the up/down arrow keys to select a worker runtime:dotnet
Use the up/down arrow keys to select a language:c#
Creating this template will make changes to existing files:
  Overwrite   ./host.json

To create the template anyway, run the command with '--force' option:
   dotnet new func --Framework net6.0 --AzureFunctionsVersion v4 --name dotnetIsolated --StorageConnectionStringValue UseDevelopmentStorage=true --force

For details on the exit code, refer to https://aka.ms/templating-exit-codes#73

Error creating project template
```

### PR DESCRIPTION
**Pull request by SatishKalluri**
…les for dotnet #4581

### Fix for - Running func new in an empty directory creates incorrect files

resolves #4581 

### Pull request checklist

* [ x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/Cli/func/Actions/LocalActions/CreateFunctionAction.cs [hunk 1]
```diff
@@ -113,6 +113,7 @@ public override async Task RunAsync()
             // Ensure that the _templates are loaded before we proceed
             _templates = await _templatesManager.Templates;
 
+            // Ensure that the worker runtime and language are set.
```
Comment (liliankasem) [inline src/Cli/func/Actions/LocalActions/CreateFunctionAction.cs:115]: The only changes in this PR is a comment, was there meant to be something else?

Comment (SatishKalluri) [reply src/Cli/func/Actions/LocalActions/CreateFunctionAction.cs:116]: Line 117 -  await UpdateLanguageAndRuntime(); also

Comment (SatishKalluri) [reply src/Cli/func/Actions/LocalActions/CreateFunctionAction.cs:115]: Yes Lilian, after reverting only this comment is the addition

Comment (liliankasem) [reply src/Cli/func/Actions/LocalActions/CreateFunctionAction.cs:115]: Sorry I don't understand how this makes sense, what does a comment have to do with the original issue?

In the issue, you mention:

>@liliankasem have checked the issue and debugged, found out that little line order change in the code to check the language and runtime been selected before creating any templates if we run 'func new' without 'func init' will resolve this issue by changing the line order of 116 prior to 114 - [https://github.com/Azure/azure-functions-core-tools/blob/main/src/Cli/func/Actions/LocalActions/CreateFunctionAction.cs](https://github.com/Azure/azure-functions-core-tools/issues/url)

This makes sense so I was expecting this PR to swap around these checks

Comment (SatishKalluri) [reply src/Cli/func/Actions/LocalActions/CreateFunctionAction.cs:115]: Hi @liliankasem ... the fix which was mentioned throwing build error for some test cases, so we are fixing it and will remove the unnecessary comments in the next fix shortly

**Comment by liliankasem** [comment]
Hi @SatishKalluri - I have a prosposed solution for this issue: https://github.com/Azure/azure-functions-core-tools/pull/4672

Do you mind working of that branch instead to ensure edge cases are accounted for and tests are write where needed?

**Comment by SatishKalluri** [comment]
Hi @liliankasem, Sure 
Wanted to confirm, can you please let me know if I can close my PR and continue working on your branch

**Comment by liliankasem** [comment]
> Hi @liliankasem, Sure Wanted to confirm, can you please let me know if I can close my PR and continue working on your branch

Yes, please do that
